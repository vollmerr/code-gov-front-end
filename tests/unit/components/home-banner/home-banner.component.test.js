import React from 'react'
import { shallow } from 'enzyme'

import HomeBanner from 'components/home-banner/home-banner.component'

const props = {
  motto: 'test-motto',
  subtitle: 'test-subtitle',
  backgroundImage: 'test-bg-image',
}

let wrapper
let instance
describe('components - HomeBanner', () => {
  beforeEach(() => {
    wrapper = shallow(<HomeBanner {...props} />)
    instance = wrapper.instance()
  })

  describe('scrollToAbout', () => {
    it('should scroll to the offset location', () => {
      // top - relies on fact that only single getElementById
      document.getElementById = () => ({ clientHeight: 100 })
      // offset - relies on fact that only single querySelector
      document.querySelector = () => ({ clientHeight: 25 })
      instance.scrollToAbout()
      expect(window.scrollTo).toBeCalledWith({ top: 75, behavior: 'smooth' })
    })
  })

  describe('render', () => {
    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })
})