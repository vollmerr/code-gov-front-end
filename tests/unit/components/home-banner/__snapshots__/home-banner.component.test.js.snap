// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components - HomeBanner render should render correctly 1`] = `
<section
  className="banner medium"
  id="banner-home"
  style={
    Object {
      "backgroundImage": "url('test-bg-image')",
    }
  }
>
  <div
    className="banner-content"
  >
    <div
      className="banner-title"
    >
      test-motto
    </div>
    <div
      className="banner-subtitle show-w-gt-1200"
    >
      test-subtitle
    </div>
  </div>
  <a
    className="scroll-indicator"
    onClick={[Function]}
    title="Scroll Down"
  >
    <i
      className="icon icon-angle-down"
    />
  </a>
</section>
`;
